lib/fs-driver.js

```diff
@@ -0,0 +1,165 @@
+'use strict';
+
+var childProcess = require('child_process');
+var last = require('101/last');
+var fs = require('fs');
+
+/**
+ * Driver for performing filesystem operations.
+ * @module fs-transform:fs-driver
+ * @author Ryan Sandor Richards
+ */
+module.exports = FsDriver;
+
+/**
+ * Creates a new FsDriver that performs operations relative to a given root
+ * directory.
+ *
+ * @example
+ * // Create a new filesystem driver relative to /tmp
+ * var driver = new FsDriver('/tmp');
+ *
+ * @class
+ * @param {string} root Path to use for operations.
+ */
+function FsDriver(root) {
+  if (!root) {
+    root = '';
+  }
+
+  if (last(root) === '/') {
+    this.root = root.substr(0, root.length - 1);
+  }
+  else {
+    this.root = root;
+  }
+}
+
+/**
+ * Makes a relative path absolute to the root directory of the driver. Returns
+ * the original path if it is already absolute.
+ *
+ * @example
+ * var driver = new FsDriver('/root/dir');
+ * // returns '/root/dir/example.txt'
+ * driver.absolutePath('example.txt');
+ *
+ * // returns '/awesome/sauce' (already an absolute path)
+ * driver.absolutePath('/awesome/sauce');
+ *
+ * @param {string} path Path to make absolute.
+ * @return {string} An absolute path relative to the root for a relative path,
+ *  or just the path if it was already absolute.
+ */
+FsDriver.prototype.absolutePath = function (path) {
+  if (path.charAt(0) === '/') {
+    return path;
+  }
+  return this.root + '/' + path;
+}
+
+/**
+ * Moves a file.
+ *
+ * @example
+ * // Has the same behavior as `mv srcfile dstfile` on the command-line.
+ * driver.move('srcfile', 'dstfile', function (err, result) {
+ *   // err will only be present if something went wrong
+ *   // result will contain information concerning the move
+ * })
+ *
+ * @param {string} source Source file path.
+ * @param {string} dest Destination file path.
+ * @param {function} cb Callback to execute with the results of the file move.
+ */
+FsDriver.prototype.move = function (source, dest, cb) {
+  var command = [
+    'mv',
+    this.absolutePath(source),
+    this.absolutePath(dest)
+  ].join(' ');
+  childProcess.exec(command, cb);
+};
+
+/**
+ * Copies a file.
+ *
+ * @example
+ * // Has the same behavior as `cp src dest` on the command-line.
+ * driver.copy('src', 'dest', function (err, result) {
+ *   // ...
+ * });
+ *
+ * @param {string} source Source file path.
+ * @param {string} dest Destination file path.
+ * @param {function} cb Callback to execute with the results of the file copy.
+ */
+FsDriver.prototype.copy = function (source, dest, cb) {
+  childProcess.exec([
```

Comment:
Yes, the use case for our purposes is to generate a shell script that can be run from a dockerfile after the user is happy with their transformations. So we want to just use regular shell commands here.